pinot-core/src/main/java/org/apache/pinot/core/query/request/context/ThreadTimer.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.query.request.context;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadMXBean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class ThreadTimer {
```

Comment:
Please add javadoc

Reply:
+1

Reply:
Make this class configurable via instance configuration. If the instance configuration asks not to measure this, then do as appropriate in in the class below, with as minimal an overhead as possible. Even though we have measured the overhead to be near-zero in high throughput use cases, it is always useful to have a mechanism to turn it off in production when we first introduce it. It will also help us roll out the feature slowly in specific clusters or data centers

Eventually, we will get rid of the config and make it standard.

Reply:
done